hazelcast/src/main/java/com/hazelcast/cluster/impl/MulticastJoiner.java

```diff
@@ -191,9 +191,8 @@ private int calculateTryCount() {
         } catch (NumberFormatException e) {
             lastDigits = RandomPicker.getInt(512);
         }
-        lastDigits = lastDigits % 100;
         int portDiff = node.getThisAddress().getPort() - networkConfig.getPort();
-        tryCount += lastDigits + portDiff * timeoutSeconds * 3;
+        tryCount += (lastDigits + portDiff) % 10;
```

Comment:
Additionally, we can change `PUBLISH_INTERVAL` to a variable interval, like between 50 ~ 200 for example, to avoid all nodes publishing at the same time, instead of a constant 100.